src/index.js

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable max-classes-per-file */
+/* eslint-disable  import/no-cycle */
+
+import './style.css';
+import interactive from './interactiveList.js';
+import clearCompletedTasks from './addRemove.js';
+
+class SingleToDo {
+  constructor(description) {
+    this.description = description;
+    this.completed = false;
+    this.index = 0;
+  }
+}
+class ToDoList {
+  constructor(toDoTasksArray = [], container) {
+    this.toDoTasksArray = toDoTasksArray;
+    this.container = document.querySelector(container);
+  }
+
+  addToDo(todo) {
+    const newToDo = new SingleToDo(todo);
+    this.toDoTasksArray.push(newToDo);
+    // update index
+    this.toDoTasksArray = this.toDoTasksArray.map((todo, index = 1) => {
+      todo.index = index;
+      return todo;
+    });
+    this.displayToDo();
+    this.setListToLocal(this.toDoTasksArray);
+    window.location.reload();
+  }
+
+  removeToDo(todoId) {
+    const filterToDo = this.toDoTasksArray.filter((todo) => parseInt((todoId), 10) !== todo.index);
+    this.toDoTasksArray = filterToDo;
+    // update index
+    this.toDoTasksArray = this.toDoTasksArray.map((todo, index = 1) => {
+      todo.index = index;
+      return todo;
+    });
+    this.displayToDo();
+    this.setListToLocal(this.toDoTasksArray);
+    window.location.reload();
+  }
+
+  setToDoArray(newToDoArray) {
+    this.toDoTasksArray = newToDoArray;
+    this.setListToLocal(this.toDoTasksArray);
+  }
+
+  getToDoArray() {
+    return this.toDoTasksArray;
+  }
+
+  setListToLocal() {
+    localStorage.setItem('lists', JSON.stringify(this.toDoTasksArray));
+  }
+
+  getListFromLocal() {
+    const getList = localStorage.getItem('lists');
+    if (getList) {
+      this.toDoTasksArray = JSON.parse(getList);
+    }
+    this.displayToDo();
+  }
+
+  displayToDo() {
+    this.toDoTasksArray.sort((a, b) => a.index - b.index);
+
+    this.container.innerHTML = this.toDoTasksArray.map((todo) => `
+        <article id=${todo.index}>
+        <div>
+        <input ${todo.completed ? 'checked' : ''} type="checkbox">
+        <textarea class = "${todo.completed ? 'complete' : ''} text-area-class" rows="1" cols="30">${todo.description}</textarea> 
+        <button id=${todo.index} class="todo-btn" > &#128465;</button></div>
+        <hr class="line-break">
+        </article>`).join('');
+  }
+}
+
+const myToDo = new ToDoList([], '.list-item');
+
+document.addEventListener('click', (e) => {
+  interactive(e);
```

Comment:
Here when I click anywhere on the page it shows an error in the console, maybe you can add a check here, to avoid errors on the console.